compiler-rt/lib/ctx_profile/RootAutoDetector.h

```diff
@@ -53,5 +54,35 @@ class PerThreadCallsiteTrie {
   /// thread, together with the number of samples that included them.
   DenseMap<uptr, uint64_t> determineRoots() const;
 };
+
+class RootAutoDetector final {
```

Comment:
nit: Just RootDetector since there aren't any other types of detectors?

Reply:
well... but the file names are this way, and the global name is `RootDetector` in CtxInstrProfiling.cpp.